pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -270,12 +278,15 @@ func (r *reconciler) reconcile(ctx context.Context, mr *resourcesv1alpha1.Manage
 		}
 	}
 
+	// get the current checksum of referenced secret data stored as annotation.
+	currentChecksum := mr.ObjectMeta.GetAnnotations()[v1beta1constants.AnnotationManagedResourceSecretChecksum]
```

Comment:
Why is an annotation used instead of a proper status field?

Reply:
We only set annotations with checksums to make `Deployment` and friends roll out the pods because of a changed template.
I don't know of any resource where we read back these checksum annotations.
In my opinion, a status field is better for this purpose.

Reply:
Done with [commit](https://github.com/gardener/gardener/pull/6136/commits/47235b9c43d325412e674b0d15b6de2c56f089db).